project/src/main/java/caselab/service/DocumentTypesService.java

```diff
@@ -1,26 +1,52 @@
 package caselab.service;
 
+import caselab.controller.types.payload.DocumentTypeDTO;
 import caselab.domain.entity.DocumentType;
 import caselab.domain.repository.DocumentTypesRepository;
 import java.util.Optional;
 import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatusCode;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
 
+@SuppressWarnings({"MagicNumber", "LineLength"})
 @Service
 @RequiredArgsConstructor
 public class DocumentTypesService {
     private final DocumentTypesRepository documentTypesRepository;
 
-    public Optional<DocumentType> findDocumentTypeById(Long id) {
-        return documentTypesRepository.findById(id);
+    public DocumentTypeDTO findDocumentTypeById(Long id) {
+        Optional<DocumentType> optionalDocumentType = documentTypesRepository.findById(id);
```

Comment:
Старайся использовать var, очень крутая штука, позволяет читаемость повысить.
```suggestion
        var optionalDocumentType = documentTypesRepository.findById(id);
```

Reply:
Не согласен. Optional тут не выглядит чем-то громоздким и непонятным, меня все устраивает. var лучше юзать в других более подходящих для этого случаях.

Reply:
> Не согласен. Optional тут не выглядит чем-то громоздким и непонятным, меня все устраивает. var лучше юзать в других более подходящих для этого случаях.

В [этой статье](https://topjava.ru/blog/26-items-for-dissecting-java-local-variable-type-inference) перечислены подходящие случаи с примерами. Конкретно с Optional<> пример приведён в 15 пункте

Reply:
Хорошо. С удовольствием изучу каждый пункт данного топика и с нетерпением буду ждать твоих пр ;)

Reply:
Я тут с var тоже не сильно согласен, Optional выглядит более понятно, как по мне